tests/basic.test.tsx

```diff
@@ -677,8 +677,11 @@ it('can use exposed types', () => {
     }
   }
   const selector: StateSelector<ExampleState, number> = (state) => state.num
-  const partial: PartialState<ExampleState> = { num: 2, numGet: () => 2 }
-  const partialFn: PartialState<ExampleState> = (state) => ({
+  const partial: PartialState<ExampleState, 'num' | 'numGet'> = {
```

Comment:
I see. Looks more precise.
BTW, we should remove `extends State` in L664.

Reply:
I wish we didn't have to do it, but without the explicit second argument it does not pass the assertion in `checkAllTypes` after our migration to `object`. I would say it even makes sense :)
At the same time, if we need to make all these changes in the library code because of the migration to `object`, there is a risk it might break code that consumes the library.
I see two ways to go from here to mitigate the risk:
1. Migrate to `object` and make it a minor release, not a patch one.
2. Revert changes related to `object` migration and keep using `Record`.

Personally, I would suggest option #1.

Reply:
Yes, this looks good and more like what I wanted. Let's go with minor, and describe breaking changes in types in the release note.